src/tests/ftest/container/per_server_fault_domain.py

```diff
@@ -0,0 +1,330 @@
+"""
+  (C) Copyright 2023 Intel Corporation.
+
+  SPDX-License-Identifier: BSD-2-Clause-Patent
+"""
+from collections import defaultdict
+from ior_test_base import IorTestBase
+from apricot import skipForTicket
+
+
+class PerServerFaultDomainTest(IorTestBase):
+    # pylint: disable=too-many-ancestors
+    """
+    Test Class Description:
+    The granularity of redundancy factor can be changed to node (rf_lvl:2). This means
+    that in a two-engine setup, if one of the ranks goes down, we count as one. If both of
+    the ranks in the same node go down, we still count as one. The default mode is engine
+    (rf_lvl:1).
+
+    If the number of ranks that has gone down exceeds the rf value, container status would
+    become UNCLEAN. For example, with an rf:1 container, if two ranks go down, the status
+    would be UNCLEAN. If one rank goes down, it would be HEALTHY.
+
+    Verify the node mode by stopping one/two ranks in one/two nodes.
+
+    Assume that the two ranks in the same node map to the same IP address.
+
+    :avocado: recursive
+    """
+
+    def verify_per_server_fault(self, create_pool, ranks_to_stop, properties,
+                                expected_status):
+        """Run the main test steps.
+
+        1. Create a pool if create_pool is True. Then create a container with given
+        container property.
+        2. Run IOR to write some data to the container.
+        3. Stop the ranks_to_stop ranks.
+        4. Wait for rebuild to finish.
+        5. Get container's health status.
+        6. Restart the stopped servers for cleanup.
+        7. Verify container's Health status is expected.
+
+        Args:
+            create_pool (bool): Whether to create a pool.
+            ranks_to_stop (str): Comma-separated ranks to stop.
+            properties (str): Container property defined in the test yaml (rf_1 or rf_2).
+            expected_status: Expected container status at the end. HEALTHY or UNCLEAN.
+        """
+        # 1. Create a pool if create_pool is True. Then create a container with given
+        # container property.
+        if create_pool:
+            self.add_pool()
+        self.container = self.get_container(pool=self.pool, create=False)
+        self.container.properties.update(properties)
+        self.container.create()
```

Comment:
`get_container` accepts any of the `*Parameter`
```suggestion
        self.container = self.get_container(pool=self.pool, properties=properties)
```

Reply:
Done. Thanks for pointing it out.